mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/task/RecomputeCurrentQuotasServiceContract.java

```diff
@@ -51,7 +51,7 @@
 
     UsersRepository usersRepository();
     SessionProvider sessionProvider();
-    MailboxManager mailboxManager();
+    StoreMailboxManager mailboxManager();
```

Comment:
what's the point having some contract tests if we enforce the way it's implemented?

Reply:
I agree on that.

However for StoreCurrentQuotaManager (the one below) I have some tests in that contract using increase (to test that it's resetting correctly when introducing inconsistencies)

increase and decrease are being declared in StoreCurrentQuotaManager, not in CurrentQuotaManager. I hesitated to move those operations to CurrentQuotaManager (inclusing resetQuotas) and remove the StoreCurrentQuotaManager, as all our implementation use it anyway. But I was unsure. WDYT?

Reply:
if you need `increase` then you must have access to it _without enforcing an implementation_.
It means it must be in some api